### PR TITLE
docs(developing): adding framework documentation for developing section

### PR DESCRIPTION
**Pull request by jeffchew**
### Related Ticket(s)

Refs #642 

### Description

This updates the `developing` section to include framework guidance, as well as updated general development documentation.

### Changelog

**New**

- Developing / Frameworks
- Other packages

**Changed**

- Developing landing page

### CODE REVIEW

File: src/pages/developing/index.mdx [hunk 1]
```diff
@@ -9,111 +9,70 @@ import { Link } from 'gatsby';
...
-
-**Note:** The expressive theme has been integrated into Carbon Design System’s core library. This means that expressive continues to be available to IBM.com, but now all Carbon adopters can get both productive and expressive experiences from the same library source. This change will be seamless for current users of Carbon for IBM.com. In the next few sprints, we will be shifting to use expressive styling and elements from core, after which the expressive theme within Carbon for IBM.com will no longer be maintained. <Link to="/help/expressive-update">Learn more about this change</Link>.
-
-</InlineNotification>
-
 <AnchorLinks>
-<AnchorLink>Carbon for IBM.com</AnchorLink>
-<AnchorLink>Get started</AnchorLink>
-<AnchorLink>Packages</AnchorLink>
+<AnchorLink>Choose Your Framework</AnchorLink>
```
Comment (janchild) [inline src/pages/developing/index.mdx]:
```suggestion
<AnchorLink>Choose your framework</AnchorLink>
```

File: src/pages/developing/index.mdx [hunk 2]
```diff
@@ -9,111 +9,70 @@ import { Link } from 'gatsby';
...
-If you are looking to build out a light static site, the
-[NextJS template](https://github.com/carbon-design-system/carbon-for-ibm-dotcom-nextjs-template) is available (using the
-React version of the Carbon for IBM.com). This includes the default configurations for the `<DotcomShell>`, as well as
-some sample components to get you started.
+Carbon for IBM.com is available in either <Link to="/developing/frameworks/web-components">web components</Link> or 
+<Link to="/developing/frameworks/react">react</Link>. Choosing a framework would depend on the needs of 
+your application. 
 
-### Available Components
+### Web Components
```
Comment (janchild) [inline src/pages/developing/index.mdx:32]:
```suggestion
### Web components
```

**Review by janchild — COMMENTED** [review]
These pages are wonderful. As I said in my comments, this makes it feel so easy. 🎉 

I made a some capitalization tweaks, that's it. It reads beautifully. On the pages where there were a few edits, I used the `Edit file` option rather than inline edits. I'm happy to work either way so if you'd prefer inline just let me know.  🎉

**Comment by jeffchew** [comment]
Thank you @janchild for reviewing and the commit suggestions!

I'm fine either way! Will get these suggestions in!

**Comment by jeffchew** [comment]
@janchild switching this to `Ready for review`! I took the suggestion of removing the parenthesis in the H2's on the web components page. 

I'm also thinking of dynamically importing this page and adding as an additional page, but can probably tackle it in a separate PR: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/master/docs/building-for-ibm-dotcom.md

**Review by janchild — APPROVED** [review]
🥂 Looks great!

**Comment by JennySanchez** [comment]
Hi @jeffchew The pages look awesome!!!

A few things for us to take another look...

- We address Web components with an only cap the "W" in the word "web" but not the "C" in "components." I see inconsistency in the Get started page as well. I know some of them are at the beginning of a sentence. Still, I thought since Web components is an application, when addressing it, we will permanently cap the "W". @janchild , can you confirm the proper format so we can share it with the team to make sure everyone is on the same page on how to address it in the future?

Artboard2

Artboard


- For the Web components page, for consistency, I wonder if the first section title should be " Install" and the "via import" and "via CDN" can be a level down under the "Install"? Also, will move "React Wrapper" section moved to be above the "Available components" works better, so it matches the anchor link order set up for the React page?

- For the IBM.com shell section, should we also include a link to the "Dotcom shell" component website page?

- Just out of curiosity, I noticed React a "Get started" section; why Web components page does not have that section?

**Comment by janchild** [comment]
@JennySanchez I can explain the capitalization for web components. When we're talking about them like a product name, we are capitalizing both words because this matches the way we refer to our other framework names. But when we are talking about web components in general, they can be treated like a common noun and they don't need to be capitalized. @jeffchew and I talked this one through when he was adding web components to the Carbon core pages (in my previous life ;) ).

**Comment by janchild** [comment]
@JennySanchez On the install headings, it could work the way you suggested but it would be faster for someone to identify their installation method at the top of the page and then click straight to the section they need. Because there isn't general content pertaining to both sections that might sit between the H2 and any H3s, I think this hierarchy works.

**Pull request merged**